TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyDecodedHttpRequestPreliminaryConvertHandler.java

```diff
@@ -136,9 +143,10 @@ private void handlePayloadNeutralTransformationOrThrow(ChannelHandlerContext ctx
     private boolean headerFieldsAreIdentical(HttpRequest request, HttpJsonMessageWithFaultingPayload httpJsonMessage) {
         if (!request.uri().equals(httpJsonMessage.path()) ||
                 !request.method().toString().equals(httpJsonMessage.method()) ||
-                request.headers().size() != httpJsonMessage.headers().strictHeadersMap.size()) {
+                request.headers().names().size() != httpJsonMessage.headers().strictHeadersMap.size()) {
             return false;
         }
+        // Depends on header size check above for correctness
```

Comment:
thanks for the comment.  That's why the size check was important and not just an optimization :)